dask/dataframe/tests/test_multi.py

```diff
@@ -262,6 +264,168 @@ def test_sequential_joins():
+    A = pd.DataFrame({'a': [1, 5, 10], 'left_val': ['a', 'b', 'c']})
+    a = dd.from_pandas(A, npartitions=2)
+    B = pd.DataFrame({'a': [1, 2, 3, 6, 7], 'right_val': [1, 2, 3, 6, 7]})
+    b = dd.from_pandas(B, npartitions=2)
+
+    C = pd.merge_asof(A, B, on='a', direction='forward').set_index('a')
+    c = merge_asof(a, b, on='a', direction='forward')
+    assert_eq(c, C)
+
+
+def test_merge_asof_on_nearest():
+    A = pd.DataFrame({'a': [1, 5, 10], 'left_val': ['a', 'b', 'c']})
+    a = dd.from_pandas(A, npartitions=2)
+    B = pd.DataFrame({'a': [1, 2, 3, 6, 7], 'right_val': [1, 2, 3, 6, 7]})
+    b = dd.from_pandas(B, npartitions=2)
+
+    C = pd.merge_asof(A, B, on='a', direction='nearest').set_index('a')
+    c = merge_asof(a, b, on='a', direction='nearest')
+    assert_eq(c, C)
+
+
+def test_merge_asof_indexed():
+    A = pd.DataFrame({'left_val': ['a', 'b', 'c']}, index=[1, 5, 10])
+    a = dd.from_pandas(A, npartitions=2)
+    B = pd.DataFrame({'right_val': [1, 2, 3, 6, 7]}, index=[1, 2, 3, 6, 7])
+    b = dd.from_pandas(B, npartitions=2)
+
+    C = pd.merge_asof(A, B, left_index=True, right_index=True)
+    c = merge_asof(a, b, left_index=True, right_index=True)
+    assert_eq(c, C)
+
+
+def test_merge_asof_on_by():
+    times_A = [parse(d) for d in ['2016-05-25 13:30:00.023',
+                                  '2016-05-25 13:30:00.023',
+                                  '2016-05-25 13:30:00.030',
+                                  '2016-05-25 13:30:00.041',
+                                  '2016-05-25 13:30:00.048',
+                                  '2016-05-25 13:30:00.049',
+                                  '2016-05-25 13:30:00.072',
+                                  '2016-05-25 13:30:00.075']]
+    tickers_A = ['GOOG', 'MSFT', 'MSFT', 'MSFT', 'GOOG', 'AAPL', 'GOOG', 'MSFT']
+    bids_A = [720.50, 51.95, 51.97, 51.99, 720.50, 97.99, 720.50, 52.01]
+    asks_A = [720.93, 51.96, 51.98, 52.00, 720.93, 98.01, 720.88, 52.03]
+    times_B = [parse(d) for d in ['2016-05-25 13:30:00.023',
+                                  '2016-05-25 13:30:00.038',
+                                  '2016-05-25 13:30:00.048',
+                                  '2016-05-25 13:30:00.048',
+                                  '2016-05-25 13:30:00.048']]
+    tickers_B = ['MSFT', 'MSFT', 'GOOG', 'GOOG', 'AAPL']
+    prices_B = [51.95, 51.95, 720.77, 720.92, 98.00]
+    quantities_B = [75, 155, 100, 100, 100]
+
+    A = pd.DataFrame({'time': times_A, 'ticker': tickers_A, 'bid': bids_A, 'ask': asks_A})
+    a = dd.from_pandas(A, npartitions=4)
+    B = pd.DataFrame({'time': times_B, 'ticker': tickers_B, 'price': prices_B, 'quantity': quantities_B})
+    b = dd.from_pandas(B, npartitions=3)
+
+    C = pd.merge_asof(B, A, on='time', by='ticker').set_index('time')
+    c = merge_asof(b, a, on='time', by='ticker')
+    assert_eq(c, C)
+
+
+def test_merge_asof_on_by_tolerance():
+    times_A = [parse(d) for d in ['2016-05-25 13:30:00.023',
```

Comment:
Use `pd.to_datetime(...)` on the list, rather than `parse(d)` on each element. Here and in the next test.